### PR TITLE
add blog post for Action development for Android devs

### PR DESCRIPTION
**Pull request by patjackson52**

### CODE REVIEW

File: _posts/blog/2017-07-19-Actions-for-android-devs.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+categories: Development
+tags: [Google Assistant, Actions on Google, conversational interfaces, Android, Kotlin]
+---
+
+
+# Google Assistant Development for Android Devs (Part 1)
+
+![](https://developers.google.com/actions/images/surfaces-hero.svg)
+
+There is a lot of buzz and excitment around the [Google Assistant Actions](https://developers.google.com/actions/), and with good reason:  this is an entirely new platform for building experiences for a number of devices.  Building a conversational Action with Google Assistant means it will be available on Android, iOS, Google Home, and others as they com available.
```
Comment (yorb) [inline _posts/blog/2017-07-19-Actions-for-android-devs.md]: Typo: "...as they **become** available..."

File: _posts/blog/2017-07-19-Actions-for-android-devs.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+
+For the uninitiated, Actions on Google are third party "Actions" that can be built using Google's APIs.  Actions extend what the Google Assistant can do by letting user talk to them.  They can be accessed with trigger words such as 'Ok Google, talk to Ticketmaster', or 'Ok Google, ask ticketmaster find Yankee's games this weekend' (Ticketmaster Action coming soon!)
```
Comment (yorb) [inline _posts/blog/2017-07-19-Actions-for-android-devs.md]: - Can we use another word besides "Actions" to explain what they are? Bots? Assistants? Apps? Scripts?
  "...Actions on Google are third-party **apps** that can be built..."
- Need a hyphen: "...**third-party**..."
- Need a period at the end of the paragraph (after parenthetical or before, your choice)

File: _posts/blog/2017-07-19-Actions-for-android-devs.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+<img src="https://storage.googleapis.com/banjotms.appspot.com/Screenshot_20170718-103901.png" width="300"> 
+
+At Ticketmaster Mobile Studio, we decided to use the languages and dev environment we are most familiar with - Kotlin/Java, IntelliJ, & Gradle.  This allows us to use the language and tools we are familiar with, and to reuse some code with Android apps.  However, the documentation and examples do not provide any Java/Kotlin.  Google provides a javascript SDK for building webhooks, however so far there is not an official Java SDK.  At the end of the day this is just json in and json out, so of course we can use any language we choose.
```
Comment (yorb) [inline _posts/blog/2017-07-19-Actions-for-android-devs.md]: - Maybe reword one of the first 2 sentences, or just replace one "familiar with" with "comfortable with" to avoid repetition.
- Avoid repetition of "however": "**Unfortunately,** the documentation and examples..."

File: _posts/blog/2017-07-19-Actions-for-android-devs.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+
+Looking through the json specs and creating response objects to match is pretty tedious, so we have decided to open source an [unofficial Kotlin/Java Actions-on-google SDK](https://github.com/TicketmasterMobileStudio/actions-on-google-kotlin).  This is an early release and will we be adding functionality, with a goal of matching the official SDK.  Currently, all the conversation components are supported.  Using the Kotlin SDK you can get up and running in no time.  And since Kotlin has great interopt with Java, you can use it from Java as well.
```
Comment (yorb) [inline _posts/blog/2017-07-19-Actions-for-android-devs.md]: Typo?: "interopt" s/b "interop"?

File: _posts/blog/2017-07-19-Actions-for-android-devs.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+title: "Google Assistant Development for Android Devs (Part 1)"
+author: Patrick Jackson
+categories: Development
+tags: [Google Assistant, Actions on Google, conversational interfaces, Android, Kotlin]
+---
+
+
+# Google Assistant Development for Android Devs (Part 1)
+
+![](https://developers.google.com/actions/images/surfaces-hero.svg)
```
Comment (yorb) [inline _posts/blog/2017-07-19-Actions-for-android-devs.md]: Maybe we can host this somewhere else to avoid hot-linking to a Google image that may change?
Also, maybe we can make a custom version and use that as the "featured image" that gets used when shared on FB etc. I think you can probably find a designer to do that for you. :D

**Review by yorb — COMMENTED** [review]
Looks awesome! Found a couple typos and added a couple suggestions.

File: _posts/blog/2017-07-19-Actions-for-android-devs.md (hunk 5 above)
Comment (patjackson52) [reply _posts/blog/2017-07-19-Actions-for-android-devs.md]: Will do.  What is needed to make it a "featured image"?  Thought fb, etc pulled from the top, or some sort of algorithm.

Comment (yorb) [reply _posts/blog/2017-07-19-Actions-for-android-devs.md]: We have it set up to pull from the `featured-image` front matter variable. There's an example in the [readme](https://github.com/TicketmasterMobileStudio/ticketmastermobilestudio.github.io#featured-image).

File: _posts/blog/2017-07-19-Actions-for-android-devs.md [hunk 6]
```diff
@@ -8,19 +8,18 @@ tags: [Google Assistant, Actions on Google, conversational interfaces, Android,
...
+There is a lot of buzz and excitment around the [Google Assistant Actions](https://developers.google.com/actions/), and with good reason:  this is an entirely new platform for building experiences for a number of devices.  Building a conversational Action with Google Assistant means it will be available on Android, iOS, Google Home, and others as they become available.
 
 
-For the uninitiated, Actions on Google are third party "Actions" that can be built using Google's APIs.  Actions extend what the Google Assistant can do by letting user talk to them.  They can be accessed with trigger words such as 'Ok Google, talk to Ticketmaster', or 'Ok Google, ask ticketmaster find Yankee's games this weekend' (Ticketmaster Action coming soon!)
+For the uninitiated, Actions on Google are third-party apps that can be built using Google's APIs.  Actions extend what the Google Assistant can do by letting user talk to them.  They can be accessed with trigger words such as 'Ok Google, talk to Ticketmaster', or 'Ok Google, ask ticketmaster find Yankee's games this weekend'. (Ticketmaster Action coming soon!)
```
Comment (mgibble) [inline _posts/blog/2017-07-19-Actions-for-android-devs.md]: I think we want to emphasize the conversational piece of this tool. How about we make the query a more natural sentence by adding more articles ...  "ask Ticketmaster to find a Yankee's game this weekend"?

File: _posts/blog/2017-07-19-Actions-for-android-devs.md [hunk 7]
```diff
@@ -8,19 +8,18 @@ tags: [Google Assistant, Actions on Google, conversational interfaces, Android,
...
 
 <img src="https://storage.googleapis.com/banjotms.appspot.com/Screenshot_20170718-103901.png" width="300"> 
 
-At Ticketmaster Mobile Studio, we decided to use the languages and dev environment we are most familiar with - Kotlin/Java, IntelliJ, & Gradle.  This allows us to use the language and tools we are familiar with, and to reuse some code with Android apps.  However, the documentation and examples do not provide any Java/Kotlin.  Google provides a javascript SDK for building webhooks, however so far there is not an official Java SDK.  At the end of the day this is just json in and json out, so of course we can use any language we choose.
+At Ticketmaster Mobile Studio, we decided to use the languages and dev environment we are most familiar with - Kotlin/Java, IntelliJ, & Gradle.  This allows us to use the language and tools we are comfortable with, and to reuse some code with Android apps.  Unfortunately, the documentation and examples are only in nodejsdo not provide any Java/Kotlin.
```
Comment (mgibble) [inline _posts/blog/2017-07-19-Actions-for-android-devs.md]: I would say "development" environment instead of "dev" environment.

also, change "nodejsdo" to "Node.js and do"

File: _posts/blog/2017-07-19-Actions-for-android-devs.md [hunk 8]
```diff
@@ -8,19 +8,18 @@ tags: [Google Assistant, Actions on Google, conversational interfaces, Android,
...
 
-
-Looking through the json specs and creating response objects to match is pretty tedious, so we have decided to open source an [unofficial Kotlin/Java Actions-on-google SDK](https://github.com/TicketmasterMobileStudio/actions-on-google-kotlin).  This is an early release and will we be adding functionality, with a goal of matching the official SDK.  Currently, all the conversation components are supported.  Using the Kotlin SDK you can get up and running in no time.  And since Kotlin has great interopt with Java, you can use it from Java as well.
+Looking through the json specs and creating response objects to match is pretty tedious, so we have decided to open source an [unofficial Kotlin/Java Actions-on-google SDK](https://github.com/TicketmasterMobileStudio/actions-on-google-kotlin).  This is an early release and will we be adding functionality, with a goal of matching the official SDK.  Currently, all the conversation components are supported.  Using the Kotlin SDK you can get up and running in no time.  And since Kotlin has great interop with Java, you can use it from Java as well.
```
Comment (mgibble) [inline _posts/blog/2017-07-19-Actions-for-android-devs.md]: I would capitalize JSON .

also, "this is an early release and we will be adding ..." change order of "will we".

also change interop to interoperability.

Comment (louis-antonopoulos) [reply _posts/blog/2017-07-19-Actions-for-android-devs.md]: "excitment"

"for a number of devices" --> "for multiple devices"

"For the uninitiated," --> I'd remove this

"can do by letting user talk" --> "can do by letting the user talk"

"Ok Google, ask ticketmaster find" --> capitalize Ticketmaster, and make it "to find"

"Yankee's games" --> no apostrophe

"we decided to use the languages and dev environment we are most familiar with - [...] This allows us to use the language and tools we are comfortable with" --> This is kind of repetitive. Also states the obvious. Maybe "leverage our expertise" or something like that?

**Comment by louis-antonopoulos** [comment]
"excitment"

"for a number of devices" --> "for multiple devices"

"For the uninitiated," --> I'd remove this

"can do by letting user talk" --> "can do by letting the user talk"

"Ok Google, ask ticketmaster find" --> capitalize Ticketmaster, and make it "to find"

"Yankee's games" --> no apostrophe

"we decided to use the languages and dev environment we are most familiar with - [...] This allows us to use the language and tools we are comfortable with" --> This is kind of repetitive. Also states the obvious. Maybe "leverage our expertise" or something like that?

**Comment by louis-antonopoulos** [comment]
"Excitment" --> Excitement

"a Yankees games" --> "a Yankees game"

"& Gradle" --> "and Gradle"

"building a Action" --> "building an Action"

File: _posts/blog/2017-07-19-Actions-for-android-devs.md [hunk 9]
```diff
@@ -0,0 +1,25 @@
+---
+title: "Google Assistant Development for Android Devs (Part 1)"
+author: Patrick Jackson
+categories: Development
+tags: [Google Assistant, Actions on Google, conversational interfaces, Android, Kotlin]
+---
+
+
+# Google Assistant Development for Android Devs (Part 1)
```
Comment (yorb) [inline _posts/blog/2017-07-19-Actions-for-android-devs.md]: Remove this line, it's redundant with the `title` front matter variable.

File: _posts/blog/2017-07-19-Actions-for-android-devs.md [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+
+![](https://storage.googleapis.com/banjotms.appspot.com/assistant_surfaces.svg)
+
+There is a lot of buzz and excitement around the [Google Assistant Actions](https://developers.google.com/actions/), and with good reason:  this is an entirely new platform for building experiences for multiple devices.  Building a conversational Action with Google Assistant means it will be available on Android, iOS, Google Home, and others as they become available.
+
```
Comment (yorb) [inline _posts/blog/2017-07-19-Actions-for-android-devs.md]: Add a read-more break here:
`<!--more-->`

Otherwise, the image will be the only thing that shows in the excerpt.

**Review by yorb — CHANGES_REQUESTED** [review]
Can you lowercase the filename too? Looks odd in the slug to have "Actions" capitalized. Also, should it be closer to the title of the article for consistency? "google-assistant-development-for-android-devs-part-1"?

Sorry I missed these document format issues earlier!
